storage/shard.go

```diff
@@ -305,6 +326,25 @@ func (s *dbShard) Flush(ps persist.Persister) error {
 // a given element. Ignore for now due to lack of proper API.
 // TODO(xichen): Propagate ticking stats back up.
 func (s *dbShard) Tick(ctx context.Context) error {
+	var (
+		numberOfLoadedSegments   int
+		numberOfUnloadedSegments int
+	)
+	if err := s.forEachSegment(func(segment sealedFlushingSegment) error {
+		if segment.LoadedStatus() == segmentUnloaded {
+			numberOfUnloadedSegments++
+		} else {
+			numberOfLoadedSegments++
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	s.metrics.numberOfLoadedSegments.Update(float64(numberOfLoadedSegments))
```

Comment:
There are 3 states: loaded, unloaded, and partially loaded. Can we have one counter for each?